cancat/uds.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+import cancat
+import threading
+import time
+
+class UDS:
+    def __init__(self, c, tx_arbid, rx_arbid):
+        self.c = c
+        self.tx_arbid = tx_arbid
+        self.rx_arbid = rx_arbid
+        self.t = threading.Thread(target = self.SendTesterPresent)
+        self.TesterPresent = False
+        self.t.setDaemon(True)
+        self.t.start()
+
+    def ReassembleIsoTP(self, start_index = 0):
+        msg_found = False
+        while msg_found is False:
+            for idx, ts, arbid, msg in self.c.genCanMsgs(start=start_index, arbids=[self.rx_arbid]):
+                msg_found = True
+                return hex(arbid), msg.encode('hex')
+            time.sleep(0.1)
+
+    def SendTesterPresent(self):
+        while True:
```

Comment:
this should end up having an internal field that specifies when to stop processing.  not urgent, but something that allows us to shut down this thread.